torchinductor/ir.py

```diff
@@ -1893,7 +1893,11 @@ def create(cls, a, b):
             size=[b3, m, n],
         ).as_fixed()
 
-        if b3 == 1:
+        if (
+            b3 == 1
+            and is_contiguous_storage_and_layout(a)
+            and is_contiguous_storage_and_layout(b)
+        ):
             # convert to normal mm
             data = MatrixMultiply(
                 layout=output_layout.as_fixed(),
```

Comment:
@jansel  I guess another option is to generate a ReinterpretView instead of View when the batch size is 1, but I am not sure if that is safe to do here.